### PR TITLE
Make persisted state pluggable

### PR DESCRIPTION
**Pull request by DaveCTurner**

### CODE REVIEW

**Review by fcofdez — APPROVED** [review]
LGTM

**Review by tlrx — APPROVED** [review]
LGTM, left a minor comment

File: server/src/main/java/org/elasticsearch/plugins/ClusterCoordinationPlugin.java [hunk 1]
```diff
@@ -33,4 +43,29 @@ default BiConsumer<DiscoveryNode, ClusterState> getJoinValidator() {
     default Map<String, ElectionStrategy> getElectionStrategies() {
         return Collections.emptyMap();
     }
+
+    default @Nullable PersistedStateFactory getPersistedStateFactory() {
```
[the remark this answers is not in the document]
Comment (tlrx) [reply server/src/main/java/org/elasticsearch/plugins/ClusterCoordinationPlugin.java]: I prefer this way, thanks

**Review by tlrx — APPROVED** [review]
LGTM (bis)

**Pull request merged**